### PR TITLE
car turning

### PR DESCRIPTION
**Pull request by allen-huang9**
car-turning

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+var currentDirection = 0;
+
+document.addEventListener('keydown', function(e) {
+  // console.log('keycode: ', e.keyCode);
+  // console.log('code: ', e.code);
+
+  if(e.code === 'KeyA') {
+    currentDirection -= 45;
+
+    $car.setAttribute('style', ' width: 100px; transform-origin: center; transform: rotate(' +
```
Comment (codingcodymiller) [inline main.js]: Instead of doing all this as inline styling, you should make CSS classes for the car, and each direction :+1:

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: main.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      direction = 0;
+    }
+
+    $car.className = 'car direction' + direction;
+
+  } else if (e.code === 'KeyS') {
+
+  } else if (e.code === 'KeyW') {
+
+  }
```
Comment (shadorki) [inline main.js]: This logic seems to be difficult and hard to read.  Less code is often not as good as readable code.

You can use words like up down left and right to dictate what direction the car should go.

File: styles.css [hunk 3]
```diff
@@ -11,3 +12,32 @@ img {
...
+  transform: rotate(225deg);
+}
+
+.direction6 {
+  transform: rotate(270deg);
+}
+
+.direction7 {
+  transform: rotate(315deg);
+}
```
Comment (shadorki) [inline styles.css]: If you want the car to move in 8 directions that is fine, however you only need four.

Numbers are harder to read, so feel free to use things like `.direction.top-right` and `direction.top-left`

Again, 4 directions would be much better for MVP.

**Review by shadorki — CHANGES_REQUESTED** [review]
You also need a data model.
image

**Review by shadorki — CHANGES_REQUESTED** [review]
Its a bit better, still very hard to read, with numbers instead of direction. However it is acceptable.

Still needs a data model though.

image

**Review by shadorki — APPROVED** [review]

**Pull request merged**